core/src/main/java/edu/wpi/grip/core/operations/composite/DistanceTransformOperation.java

```
@@ -0,0 +1,124 @@
+    }
+
+    private enum MaskSize {
+
+        ZERO("0x0", 0),
+        THREE("3x3", 3),
+        FIVE("5x5", 5);
+
+        private final String label;
+        private final int value;
+
+        private MaskSize(String label, int value) {
+            this.label = label;
+            this.value = value;
+        }
+
+        @Override
+        public String toString() {
+            return label;
+        }
+    }
+
+    private final SocketHint<Mat> srcHint = SocketHints.Inputs.createMatSocketHint("Input", false);
+    private final SocketHint<Type> typeHint = SocketHints.createEnumSocketHint("Type", Type.DIST_L2);
+    private final SocketHint<MaskSize> maskSizeHint = SocketHints.createEnumSocketHint("Mask size", MaskSize.ZERO);
+
+    private final SocketHint<Mat> outputHint = SocketHints.Inputs.createMatSocketHint("Output", true);
+
+    @Override
+    public String getName() {
+        return "Distance Transform";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Performs a distance transform on an image";
+    }
+
+    @Override
+    public Category getCategory() {
+        return Category.IMAGE_PROCESSING;
+    }
+
+    @Override
+    public Optional<InputStream> getIcon() {
+        return Optional.of(getClass().getResourceAsStream("/edu/wpi/grip/ui/icons/opencv.png"));
+    }
+
+    @Override
+    public InputSocket<?>[] createInputSockets(EventBus eventBus) {
+        return new InputSocket<?>[]{
+            new InputSocket<>(eventBus, srcHint),
+            new InputSocket<>(eventBus, typeHint),
+            new InputSocket<>(eventBus, maskSizeHint)
+        };
+    }
+
+    @Override
+    public OutputSocket<?>[] createOutputSockets(EventBus eventBus) {
+        return new OutputSocket<?>[]{
+            new OutputSocket<>(eventBus, outputHint)
+        };
+    }
+
+    @Override
+    public void perform(InputSocket<?>[] inputs, OutputSocket<?>[] outputs) {
+        final Mat input = (Mat) inputs[0].getValue().get();
```

Since `input` has to be grayscale, you should throw an exception if it's not.  Otherwise you get a confusing OpenCV stack trace in the GUI